core/src/core/transaction.rs

```diff
@@ -343,10 +346,14 @@ impl Transaction {
 	/// Validates all relevant parts of a fully built transaction. Checks the
 	/// excess value against the signature as well as range proofs for each
 	/// output.
+	// TODO: this is Consensus critical code. Move to consensus.rs ?
```

Comment:
A lot of code is consensus critical in `core` and pretty much all `pipe.rs` is as well. The idea with `consensus.rs` is more to hold constants and limits, as well as simple logic. But we can't fit 1/4 of the codebase in there :-)

Reply:
Ok, removed for now. Breathing space until testnet2 might be good, alas habits will form. Soon enough it might become necessary to extract separate .rs files containing the consensus rules and unit tests that ensure that consensus isn't broken by some code refactoring. I realize that forks can be caused by almost anything. ”Le battement d'ailes d'un papillon au Brésil peut-il provoquer ...” - a bit of fun for the French among us